src/nvim/fileio.c

```diff
@@ -4333,108 +4333,95 @@ void shorten_fnames(int force)
   redraw_tabline = TRUE;
 }
 
-/*
- * add extension to file name - change path/fo.o.h to path/fo.o.h.ext
- *
- * Assumed that fname is a valid name found in the filesystem we assure that
- * the return value is a different name and ends in 'ext'.
- * "ext" MUST be at most 4 characters long if it starts with a dot, 3
- * characters otherwise.
- * Space for the returned name is allocated, must be freed later.
- * Returns NULL when out of memory.
- */
-char_u *
-modname (
-    char_u *fname,
-    char_u *ext,
-    int prepend_dot                /* may prepend a '.' to file name */
-)
+/// add extension to file name - change path/fo.o.h to path/fo.o.h.ext
+///
+/// Assumed that fname is a valid name found in the filesystem we assure that
+/// the return value is a different name and ends in 'ext'.
+/// "ext" MUST be at most 4 characters long if it starts with a dot, 3
+/// characters otherwise.
+/// Space for the returned name is allocated, must be freed later.
+/// Returns NULL when out of memory.
+/// @param fname When NULL use the current directory name
+/// @param ext The extension to add to the filename, 4 chars max if prefixed 
+///            with a dot, 3 otherwise.
+/// @param prepend_dot If true, prefix the ext with a dot
```

Comment:
When we change to new doc style, we should try to improve it if possible.  This is, new-doc-style acts as a kind of marker that doc has been "reviewed for improvements". This one was a bit tricky, but I ended with
```c
/// Get new filename ended by given extension.
///
/// @param fname        The original filename.
///                     If NULL, use current directory name and ext to
///                     compute new filename.
/// @param ext          The extension to add to the filename.
///                     4 chars max if prefixed with a dot, 3 otherwise.
/// @param prepend_dot  If true, prefix ext with a dot.
///                     Does nothing if ext already starts with a dot, or
///                     if fname is NULL.
///
/// @return [allocated] - A new filename, made up from:
///                       * fname + ext, if fname not NULL.
///                       * current dir + ext, if fname is NULL.
///                         On Windows, and if ext starts with ".", a "_" is
///                         preprended to ext (for filename to be valid).
///                       Result is guaranteed to:
///                       * be ended by <ext>.
///                       * have a basename with at most BASENAMELEN chars:
///                         original basename is truncated if necessary.
///                       * be different than original: basename chars are
///                         replaced by "_" if necessary. If that can't be done
///                         because truncated value of original filename was
///                         made of all underscores, replace first "_" by "v".
///                     - NULL, if fname is NULL and there was a problem trying
///                       to get current directory.
```